superbench/analyzer/data_diagnosis.py

```diff
@@ -93,25 +94,53 @@ def _get_baseline_of_metric(self, baseline, metric):
                 logger.warning('DataDiagnosis: get baseline - {} baseline not found'.format(metric))
                 return -1
 
-    def _get_criteria(self, rule_file, baseline_file):
-        """Get and generate criteria of metrics.
+    def __get_metrics_in_criteria(self, rule, benchmark_rules, baseline):
+        """Get metrics with baseline in the rule.
 
-        Read rule file and baseline file. For each rule, use metric with regex
-        in the metrics of the rule to match the metric full name from raw data
-        for each benchmark in the rule, and then merge baseline and rule for
-        matched metrics.
+        Use metric with regex in the metrics of the rule to match
+        the metric full name from raw data for each benchmark in the rule,
+        and then merge baseline for each metric.
```

Comment:
Parse metric regex in the rule, and store the (baseline, metric) pair in  `_sb_rules[rule]['metrics'][metric]` and metric in `_enable_metrics`

Reply:
ok